scrabble.js

```diff
@@ -1,8 +1,167 @@
+          (arrayOfWords[i].length < highest[0].length && highest[0].length != 7)) {
+            highest = [arrayOfWords[i], score];
+      }
+    }
+  }
+  return highest[0];
+}
+
+// Testing:
+console.log(Scrabble.highestScoreFrom(['dq', 'ZZZZZZ', 'turtle', 'AAAAAAF', 'QQQQQQ', 'bob', 'FAAAAAA']));
+console.log("Score for Q: " + Scrabble.scoreLetter('q'));
+
+// ==============PLAYER 'CLASS'================
+
+function Player(name) {
+  this.name = name;
+  this.plays = [];
+}
+
+Player.prototype.hasWon = function() {
+  return this.totalScore() > 100;
+}
+
+Player.prototype.totalScore = function() {
+  var total = 0;
+  for (var i = 0; i < this.plays.length; i++){
+    total += Scrabble.score(this.plays[i]);
+  }
+  return total;
+}
+
+Player.prototype.highestScoringWord = function() {
+  return Scrabble.highestScoreFrom(this.plays);
+}
+
+Player.prototype.highestWordScore = function() {
+  return Scrabble.score(Scrabble.highestScoreFrom(this.plays));
+}
+
+Player.prototype.play = function(word) {
+  if (this.hasWon()){
+    return false;
+  } else {
+    this.plays.push(word);
+  }
+}
+
+// Testing:
+bob = new Player('Bob');
+console.log(bob);
+bob.play('aaaa');
+bob.play('aaaaaaa');
+bob.play('aaa');
+console.log(bob);
+console.log("Bob has already won: " + bob.hasWon());
+bob.play('QQQQQQQ');
+console.log(bob);
+console.log("Bob has already won: " + bob.hasWon());
+bob.play('aaaaaaf');
+console.log(bob);
+console.log("Bob has already won: " + bob.hasWon());
+
+console.log(bob.highestScoringWord());
+console.log(bob.highestWordScore());
+console.log(bob.totalScore());
+console.log(bob.plays);
+
+// ==============TILEBAG 'CLASS'================
+
+function Tilebag (){
+  this.tiles = ['J', 'K', 'Q', 'X', 'Z',
+                'B', 'B',
+                'C', 'C',
+                'F', 'F',
+                'H', 'H',
+                'M', 'M',
+                'P', 'P',
+                'V', 'V',
+                'W', 'W',
+                'Y', 'Y',
+                'G', 'G', 'G',
+                'D', 'D', 'D', 'D',
+                'L', 'L', 'L', 'L',
+                'S', 'S', 'S', 'S',
+                'U', 'U', 'U', 'U',
+                'T', 'T', 'T', 'T', 'T', 'T',
+                'R', 'R', 'R', 'R', 'R', 'R',
+                'N', 'N', 'N', 'N', 'N', 'N',
+                'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O',
+                'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A',
+                'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I',
+                'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E'];
+}
+
+Tilebag.prototype.drawTiles = function(num){
+  if (num > this.tiles.length){
+    return false; // You are trying to draw more tiles than the tilebag contains
+  } else {
+    tilesPicked = []
+    for (var i = 0; i < num; i++){
+      var chosen = Math.floor(Math.random() * this.tiles.length);
+      tilesPicked.push(this.tiles.splice(chosen, 1)[0]);
+    }
```

Comment:
Good use of both splice & push to take a letter out of the tilebag and put it into the tilesPicked.